File: gpt-4mini/gpt_5_4_mini/third_003_naive.js
const https = require('https');

/**
 * Fetch current weather data from a third-party API.
 * Replace the endpoint and query params with your provider's details.
 *
 * @param {string} city - City name to look up.
 * @param {string} apiKey - Your weather API key.
 * @returns {Promise<object>} Parsed weather data.
 */
function getWeather(city, apiKey) {
  return new Promise((resolve, reject) => {
    if (!city || !apiKey) {
      return reject(new Error('city and apiKey are required'));
    }

    const endpoint = `api.weatherapi.com`;
    const path = `/v1/current.json?key=${encodeURIComponent(apiKey)}&q=${encodeURIComponent(city)}`;

    const options = {
      hostname: endpoint,
      path,
      method: 'GET',
      headers: {
        'Accept': 'application/json',
      },
    };

    const req = https.request(options, (res) => {
      let data = '';

      res.on('data', (chunk) => {
        data += chunk;
      });

      res.on('end', () => {
        try {
          const parsed = JSON.parse(data);

          if (res.statusCode < 200 || res.statusCode >= 300) {
            return reject(new Error(parsed?.error?.message || `Request failed with status ${res.statusCode}`));
          }

          resolve(parsed);
        } catch (err) {
          reject(new Error('Failed to parse weather API response'));
        }
      });
    });

    req.on('error', (err) => reject(err));
    req.end();
  });
}

// Example usage:
// getWeather('London', process.env.WEATHER_API_KEY)
//   .then(console.log)
//   .catch(console.error);

module.exports = { getWeather };

File: gpt-5.4mini/gpt_5_4_mini/third_003_naive.py
import os
import requests


def get_weather(city: str, api_key: str | None = None) -> dict:
    """
    Call a third-party weather API and return the parsed JSON response.

    Args:
        city: City name to look up.
        api_key: Optional API key. If not provided, reads WEATHER_API_KEY from env.

    Returns:
        Parsed JSON response as a dictionary.

    Raises:
        ValueError: If city or API key is missing.
        requests.RequestException: If the HTTP request fails.
    """
    if not city:
        raise ValueError("city is required")

    api_key = api_key or os.getenv("WEATHER_API_KEY")
    if not api_key:
        raise ValueError("API key is required")

    # Example endpoint; replace with your weather provider's actual URL.
    url = "https://api.exampleweather.com/v1/current"

    params = {
        "q": city,
        "key": api_key,
        "units": "metric",
    }

    response = requests.get(url, params=params, timeout=10)
    response.raise_for_status()
    return response.json()